src/net.c

```diff
@@ -179,6 +179,7 @@ struct mg_connection *mg_connect(struct mg_mgr *mgr, const char *url,
     c->is_client = true;
     c->fd = (void *) (size_t) -1;  // Set to invalid socket
     c->fn_data = fn_data;
+    c->recv_max = MG_MAX_RECV_SIZE;
```

Comment:
This seems to work well since then each subhandler can dynamically modify it as needed.